control-announcements/src/main/java/org/oskari/announcements/actions/AnnouncementsServiceMybatisImpl.java

```diff
@@ -0,0 +1,131 @@
+package org.oskari.announcements.actions;
+
+import javax.sql.DataSource;
+import org.apache.ibatis.mapping.Environment;
+import org.apache.ibatis.session.Configuration;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.apache.ibatis.session.SqlSessionFactoryBuilder;
+import org.apache.ibatis.transaction.TransactionFactory;
+import org.apache.ibatis.transaction.jdbc.JdbcTransactionFactory;
+import fi.nls.oskari.annotation.Oskari;
+import fi.nls.oskari.db.DatasourceHelper;
+import org.oskari.announcements.helpers.Announcement;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import org.json.JSONObject;
+import org.oskari.announcements.helpers.AnnouncementsParser;
+import org.oskari.announcements.mappers.AnnouncementsMapper;
+
+@Oskari
+public class AnnouncementsServiceMybatisImpl extends AnnouncementsService{
+
+    private static final Logger LOG = LogFactory.getLogger(AnnouncementsServiceMybatisImpl.class);
+
+    private SqlSessionFactory factory = null;
+    private AnnouncementsParser parser = new AnnouncementsParser();
+
+    public AnnouncementsServiceMybatisImpl() {
+        final DatasourceHelper helper = DatasourceHelper.getInstance();
+        DataSource dataSource = helper.getDataSource();
+        if (dataSource == null) {
+            dataSource = helper.createDataSource();
+        }
+        if (dataSource == null) {
+            LOG.error("Couldn't get datasource for oskari announcements service");
+        }
+        factory = initializeMyBatis(dataSource);
+    }
+
+    private SqlSessionFactory initializeMyBatis(final DataSource dataSource) {
+        final TransactionFactory transactionFactory = new JdbcTransactionFactory();
+        final Environment environment = new Environment("development", transactionFactory, dataSource);
+
+        final Configuration configuration = new Configuration(environment);
+        configuration.getTypeAliasRegistry().registerAlias(AnnouncementsService.class);
+        configuration.setLazyLoadingEnabled(true);
+        configuration.addMapper(AnnouncementsMapper.class);
+
+        return new SqlSessionFactoryBuilder().build(configuration);
+    }
+
+    public JSONObject getAdminAnnouncements() {
+        LOG.debug("get admin announcements");
+        final SqlSession session = factory.openSession();
+        JSONObject data = new JSONObject();
+        try {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            data = parser.parseAnnouncementsMap(mapper.getAdminAnnouncements());
+            session.commit();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to get admin announcements", e);
+        } finally {
+            session.close();
+        }
```

Comment:
Use try-with-resources:
```suggestion
        try (final SqlSession session = factory.openSession()) {
            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
            return parser.parseAnnouncementsMap(mapper.getAdminAnnouncements());
        } catch (Exception e) {
            throw new ServiceRuntimeException("Failed to get admin announcements", e);
        }
```